apps/demo/app/custom-ui/[...puckPath]/client.tsx

```diff
@@ -301,23 +306,106 @@ const CustomActionBar = ({ children, label }) => {
   );
 };
 
+const HeroRender = ({ props }) => {
+  const { align, title, description, buttons, padding, image, puck } = props;
+  // Empty state allows us to test that components support hooks
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const [_] = useState(0);
+  const [isExploding, setIsExploding] = useState(false);
+
+  const { getPermissions } = usePuck();
+  const permissions = getPermissions();
+
+  useEffect(() => {
+    setIsExploding(permissions?.party);
+  }, [permissions?.party]);
+
+  return (
+    <Section
+      padding={padding}
+      className={getClassName({
+        left: align === "left",
+        center: align === "center",
+        hasImageBackground: image?.mode === "background",
+      })}
+    >
+      {isExploding && <ConfettiExplosion />}
```

Comment:
🤣 as much as I love this, I think we should introduce a simpler `resolvePermissions` implementation. The easiest thing is probably to set one of the default permissions to false on another component (I vote disabled duplication on <Stats> in the custom-ui demo).